test/spec/common/index/interface.js

```diff
@@ -22,9 +22,6 @@ test('AlgoliaSearch index API spec', function(t) {
   var expectedMethods = [
     'addObject',
     'addObjects',
-    'addApiKey',
```

Comment:
tests will fail if you remove it here, this is testing that everything which is present on the prototype is actually listed here. 

Not a particularly useful test, but still. 

What you can do for clarity is: 

```js
var deprecatedMethods = [ ... ];
var validMethods = [ ... ];
var expectedMethods = validMethods.concat(deprecatedMethods);
```

Reply:
If this is the only tests we have, I think this PR is indeed not useful cc @julienbourdeau. Let's keep the tests intact until we actually remove the methods in a next major version.

WDYT? @julienbourdeau @Haroenv